vault/expiration.go

```diff
@@ -1157,6 +1160,11 @@ func (m *ExpirationManager) revokePrefixCommon(ctx context.Context, prefix strin
 func (m *ExpirationManager) Renew(ctx context.Context, leaseID string, increment time.Duration) (*logical.Response, error) {
 	defer metrics.MeasureSince([]string{"expire", "renew"}, time.Now())
 
+	// Acquire lock for this lease
+	leaseLock := m.lockForLeaseID(leaseID)
+	leaseLock.Lock()
+	defer leaseLock.Unlock()
+
 	// Load the entry
 	le, err := m.loadEntry(ctx, leaseID)
```

Comment:
since we're dealing with races between revoke and renew, do we need to check for `le != nil` here? What if we revoke just before a renew (deleting it so it can't be loaded), and then renew acquires the lock and goes through this code?

Reply:
Is leaking memory your concern here? The loadEntry code path will delete the lock if the leaseID doesn't exist.

Reply:
i was concerned about trying to renew le if it had been deleted by a revoke call that came in just before it. i just checked the code for `*leaseEntry.renewable()` though and it returns an error if `le == nil`, so i think we're good here